### PR TITLE
Citizen testing support

### PR DESCRIPTION
**Pull request by sealDebolina**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the keep-helm label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/app/case/C100CaseApi.ts [hunk 1]
```diff
@@ -42,6 +42,30 @@ export class CaseApi {
     }
   }
 
+  public async createCaseTestingSupport(): Promise<UpdateCaseResponse> {
+    try {
+      const response = await this.axios.post<UpdateCaseResponse>('/testing-support/create-dummy-citizen-case');
+      //const { id, caseTypeOfApplication, c100RebuildReturnUrl, state, noOfDaysRemainingToSubmitCase } = response?.data;
+      //console.log(response)
```
Comment (AlokDatta) [inline src/main/app/case/C100CaseApi.ts:49]: remove commented out code

File: src/main/app/case/C100CaseApi.ts [hunk 2]
```diff
@@ -42,6 +42,30 @@ export class CaseApi {
...
+      this.logError(err);
+      throw new Error('Case could not be created.');
+    }
+  }
+  public async deleteCaseTestingSupport(id: string): Promise<UpdateCaseResponse> {
+    console.log('id is' + id);
+    try {
+      const response = await this.axios.post<UpdateCaseResponse>('/testing-support/delete-citizen-case', id);
+      //const { id, caseTypeOfApplication, c100RebuildReturnUrl, state, noOfDaysRemainingToSubmitCase } = response?.data;
+      //console.log(response)
```
Comment (AlokDatta) [inline src/main/app/case/C100CaseApi.ts:61]: remove commented out code

File: src/main/app/testingsupport/TSDraftController.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import autobind from 'autobind-decorator';
+import type { Response } from 'express';
+
+import {
+  //C100_CASE_NAME,
```
Comment (AlokDatta) [inline src/main/app/testingsupport/TSDraftController.ts:5]: remove commented out code

File: src/main/app/testingsupport/TSDraftController.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+export class TSDraftController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+
+  public async post(req: AppRequest, res: Response): Promise<void> {
+    this.redirect(req, res, HOME_URL);
+  }
+
+  public async createC100Draft(req: AppRequest<AnyObject>, res: Response): Promise<void> {
```
Comment (AlokDatta) [inline src/main/app/testingsupport/TSDraftController.ts:25]: change this method name to showcase testing support

File: src/main/app/testingsupport/TSDraftController.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      req.session.userCase = (await req.locals.C100Api.retrieveCaseById(newCaseId)) as CaseWithId;
+      req.session.save(() => {
+        res.redirect(C100_CHECK_YOUR_ANSWER);
+      });
+    } catch (e) {
+      throw new Error('C100case could not be created');
+    }
+  }
+
+  public async deleteC100Draft(req: AppRequest<AnyObject>, res: Response): Promise<void> {
```
Comment (AlokDatta) [inline src/main/app/testingsupport/TSDraftController.ts:37]: change this method name to showcase testing support

File: src/main/app/testingsupport/TSDraftController.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    const data = req.body['ids'] as string;
+    const value = data.split(',');
+    value.forEach(async element => {
+      try {
+        const caseData = await req.locals.C100Api.retrieveCaseById(element);
+        await req.locals.C100Api.updateCase(element, caseData, HOME_URL, C100_CASE_EVENT.DELETE_CASE);
```
Comment (AlokDatta) [inline src/main/app/testingsupport/TSDraftController.ts:43]: probably a dummy case data will suffice

File: src/main/routes.ts [hunk 7]
```diff
@@ -217,6 +229,12 @@ export class Routes {
           this.routeGuard.bind(this, step, 'post'),
           errorHandler(new postController(step.form.fields).post)
         );
+
+        const tsDraftController = new TSDraftController(step.form.fields);
+        app.post(CREATE_DRAFT, errorHandler(tsDraftController.post));
+        app.post(`${CREATE_DRAFT}/createC100Draft`, errorHandler(tsDraftController.createC100Draft));
+        app.post(`${CREATE_DRAFT}/deleteC100Draft`, errorHandler(tsDraftController.deleteC100Draft));
```
Comment (AlokDatta) [inline src/main/routes.ts:236]: get this verified

File: src/main/routes.ts [hunk 8]
```diff
@@ -217,6 +228,12 @@ export class Routes {
           this.routeGuard.bind(this, step, 'post'),
           errorHandler(new postController(step.form.fields).post)
         );
+
+        const tsDraftController = new TSDraftController(step.form.fields);
```
Comment (vivek-sekhar) [inline src/main/routes.ts:232]: Why is this inside the loop?

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/routes.ts [hunk 9]
```diff
@@ -155,6 +166,11 @@ export class Routes {
     app.get(C100_RETRIVE_CASE, errorHandler(new GetCaseController().getC100ApplicantCase));
     app.get(C100_DOWNLOAD_APPLICATION, errorHandler(new ApplicationDownloadController().download));
 
+    const tsDraftController = new TSDraftController();
```
Comment (vivek-sekhar) [inline src/main/routes.ts:169]: Instead you can export an instance of the TSDraftController from the class file itself and use it here.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

**Review by AlokDatta — APPROVED** [review]

**Review by vivek-sekhar — APPROVED** [review]

**Pull request merged**